app/task/task.c

```diff
@@ -0,0 +1,71 @@
+/**
+  This file encapsulates the SDK-based task handling for the NodeMCU Lua firmware.
+ */
+#include "task/task.h"
+#include "mem.h"
+#include "c_stdio.h"
+
+#define TASK_HANDLE_MONIKER 0x68680000
+#define TASK_HANDLE_MASK    0xFFF80000
+#define TASK_HANDLE_UNMASK  (~TASK_HANDLE_MASK)
+#define TASK_HANDLE_SHIFT   2
+#define TASK_HANDLE_ALLOCATION_BRICK 4   // must be a power of 2
+#define TASK_DEFAULT_QUEUE_LEN 8
+
+#define CHECK(p,v,msg) if (!(p)) { NODE_DBG ( msg ); return (v); }
+
+/*
+ * Private arrays to hold the 3 event task queues and the dispatch callbacks
+ */ 
+LOCAL os_event_t *task_Q[3];
+LOCAL task_callback_t *task_func;
+LOCAL int task_count;
+ 
+LOCAL void task_dispatch (os_event_t *e) {
+  task_handle_t handle = e->sig;
+  if ( (handle & TASK_HANDLE_MASK) == TASK_HANDLE_MONIKER) {
+    uint16 entry    = (handle & TASK_HANDLE_UNMASK) >> TASK_HANDLE_SHIFT;
+    uint8  priority = handle & 3;
```

Comment:
Perhaps turn the 3 into a TASK_HANDLE_PRIO_MASK? And have a TASK_HANDLE_PRIO_MAX for the 3 on the next line?